lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -616,8 +617,12 @@ public TokenDetails requestToken(TokenParams params, AuthOptions tokenOptions) t
                     signedTokenRequest = (TokenRequest)authCallbackResponse;
                 else
                     throw AblyException.fromErrorInfo(new ErrorInfo("Invalid authCallback response", 400, 40000));
-            } catch(AblyException e) {
-                throw AblyException.fromErrorInfo(e, new ErrorInfo("authCallback failed with an exception", 401, 80019));
+            } catch (Exception e) {
+                boolean isTokenExceptionNonRetriable = e instanceof NonRetriableTokenException;
+                boolean isAblyExceptionNonRetriable = e instanceof AblyException && ((AblyException) e).errorInfo.statusCode == 403;
+                boolean shouldNotRetryAuthOperation = isTokenExceptionNonRetriable || isAblyExceptionNonRetriable;
+                int statusCode = shouldNotRetryAuthOperation ? 403 : 401; // RSA4c & RSA4d
```

Comment:
maybe we should name this resolvedStatusCode as we determine which code it should be. 
Also, should we extract these common HTTP error codes to HttpConstants and use them from there rather than have it hard coded?

Reply:
We agreed to move this to a new task which will be done later. https://github.com/ably/ably-java/issues/837